src/server.ts

```diff
@@ -259,18 +284,40 @@ files.on(EventStream.Diagnostics, "^package\\.json$", (uri, name, contents) => {
             connection.sendDiagnostics({uri: uri, diagnostics: diagnostics});
         });
         const regexVersion = new RegExp(/^(\d+\.)?(\d+\.)?(\d+)$/);
+        let request_payload = [];
         for (let dependency of deps) {
-            if(dependency.name.value && dependency.version.value && regexVersion.test(dependency.version.value)) {
-                get_metadata('npm', dependency.name.value, dependency.version.value, (response) => {
-                    if (response != null) {
-                        let pipeline = new DiagnosticsPipeline(DiagnosticsEngines, dependency, config, diagnostics);
-                        pipeline.run(response);
-                    }
-                    aggregator.aggregate(dependency);
-                });
-            } else {
-                aggregator.aggregate(dependency);
+            request_payload.push({
+                "ecosystem": "npm",
+                "package": dependency.name.value,
+                "version": dependency.version.value
+            })
+        }
+        for (let i = 0; i < request_payload.length; i = i + 10) {
+            let pck = request_payload.slice(i, i + 10);
+            let req_data = JSON.stringify(pck);
+            const options = {
+                url: url,
+                body: req_data,
+                headers: {
+                            'Content-Type': 'application/json',
+                            'Authorization' : 'Bearer ' + key,
+                        }
             }
+            postService(options).then((res) => {

```

Comment:
like `postService(options).then` success scenario, handle error scenario as well for `reject`